### ISSUE TITLE
SBT plugin published to wrong location in Maven Central

### ISSUE THREAD

**Issue by fiadliel** [flyway/flyway#1276]
###### What version of Flyway are you using?
4.0

###### What did you do?
(Please include the content causing the issue, any relevant configuration settings, and the command you ran)
Looked at the contents of https://repo1.maven.org/maven2/org/flywaydb/flyway-sbt/

Added the line
```scala
addSbtPlugin("org.flywaydb" % "flyway-sbt" % "4.0")
```
to verify the path that SBT searches for the plugin.

###### What did you expect to see?
I would expect an SBT plugin for SBT 0.13.x to encode both the scala version and SBT version in the path (for ABI reasons): https://repo1.maven.org/maven2/org/flywaydb/flyway-sbt_2.10_0.13/4.0/flyway-sbt-4.0.jar

And SBT would download/use the plugin when given the above configuration.

###### What did you see instead?
The path to (for example 4.0) is https://repo1.maven.org/maven2/org/flywaydb/flyway-sbt/4.0/flyway-sbt-4.0.jar

This is not accessed, and the plugin cannot be found.

**Comment by sourcesmith** [comment]
Place the following in your plugins config for as a work around:

```
libraryDependencies ++= Seq(
  "org.flywaydb" % "flyway-core" % "4.0",
  "org.flywaydb" % "flyway-sbt" % "4.0"
)
```

**Comment by axelfontaine** [comment]
Maven Central does not support the path structure you suggest. Hence our instructions with an additional resolver: https://flywaydb.org/documentation/sbt/

**Comment by sourcesmith** [comment]
The additional repo as of a week or two ago did not work and visiting [https://flywaydb.org/repo](https://flywaydb.org/repo) redirects to the project's home page.

**Comment by fiadliel** [comment]
I have little familiarity with Maven Central, but sbt-sonatype does publish there:
https://repo1.maven.org/maven2/org/xerial/sbt/sbt-sonatype_2.10_0.13/1.1/

I also found a Stack Overflow document of minimal help:
http://stackoverflow.com/questions/25192749/how-to-publish-sbt-plugin-to-nexus-repositories

In practice, most of the community publishes to Bintray instead, this is where I publish SBT plugins, so I wasn't aware of any extra issues publishing to Maven Central. This is the suggested route to follow, and the default resolver settings for SBT include Bintray:
http://www.scala-sbt.org/0.13/docs/Bintray-For-Plugins.html

If you fix your own custom repo, that will help, but it does still cause extra steps before use, especially for people who use an internal Sonatype Nexus as a proxy.

**Comment by fiadliel** [comment]
But fixing your private repo would be appreciated anyway, thanks :-)

**Comment by vpavkin** [comment]
+1, can't resolve flyway-sbt plugin

**Comment by ksit** [comment]
+1, is there a workaround for the time being?

**Comment by ksit** [comment]
@sourcesmith I already tried your workaround above (i.e. defining libraryDependencies in plugins.sbt), but SBT still gave me the same error - it tried to resolve the plugin using the following path:
`org/flywaydb/flyway-sbt_2.10_0.13/4.0/flyway-sbt-4.0.jar`.  If you know a different workaround to the issue, please let me know - for the time being, I just switched to use the command line version.

Note: I am using a local artifactory server as the local proxy for jcenter and maven central.

**Comment by sourcesmith** [comment]
@ksit You didn't mention that though.  I am currently using the workaround I described.  Did you remove the call to addSbtPlugin because that is responsible for adding the scala+sbt version suffix to the artifact ID in the path?  This assumes you are using the default resolver patterns, as well.

**Comment by axelfontaine** [comment]
I just tried this and everything is working as expected. I updated SBT to the latest version (0.13.11) No libraryDeependencies, no other workarounds. I wiped all caches on my machine and followed the getting started tutorial (https://flywaydb.org/getstarted/firststeps/sbt) and all downloads completed successfully:

```
C:\Temp\flyway-sbt-test>sbt flywayInfo
Java HotSpot(TM) 64-Bit Server VM warning: ignoring option MaxPermSize=256m; support was removed in 8.0
[info] Loading project definition from C:\Temp\flyway-sbt-test\project
[info] Updating {file:/C:/Temp/flyway-sbt-test/project/}flyway-sbt-test-build...
[info] Resolving org.fusesource.jansi#jansi;1.4 ...
[info] downloading https://flywaydb.org/repo/org/flywaydb/flyway-sbt_2.10_0.13/4.0/flyway-sbt-4.0.jar ...
[info]  [SUCCESSFUL ] org.flywaydb#flyway-sbt;4.0!flyway-sbt.jar (949ms)
[info] downloading https://repo1.maven.org/maven2/org/flywaydb/flyway-core/4.0/flyway-core-4.0.jar ...
[info]  [SUCCESSFUL ] org.flywaydb#flyway-core;4.0!flyway-core.jar (371ms)
[info] Done updating.
[info] Set current project to flyway-sbt-test (in build file:/C:/Temp/flyway-sbt-test/)
[info] Flyway 4.0 by Boxfuse
[info] Database: jdbc:h2:file:./target/foobar (H2 1.4)
[warn] Unable to resolve location filesystem:src/main/resources/db/migration
[warn] Unable to resolve location filesystem:src/main/resources/db/migration
[warn] Unable to resolve location filesystem:src/main/resources/db/migration
[info] +---------+-------------+---------------------+---------+
[info] | Version | Description | Installed on        | State   |
[info] +---------+-------------+---------------------+---------+
[info] | No migrations found                                   |
[info] +---------+-------------+---------------------+---------+
[success] Total time: 0 s, completed 14.04.2016 21:03:13```

Please provide the exact steps to reproduce the issue as I cannot get this to fail.

**Comment by sourcesmith** [comment]
Okay, I have cleared my cache and re-instated the configuration from the getting first steps guide and it now appears to be working:

```
[info] downloading https://flywaydb.org/repo/org/flywaydb/flyway-sbt_2.10_0.13/4.0/flyway-sbt-4.0.jar ...
[info] 	[SUCCESSFUL ] org.flywaydb#flyway-sbt;4.0!flyway-sbt.jar (583ms)
[info] downloading https://repo1.maven.org/maven2/org/flywaydb/flyway-core/4.0/flyway-core-4.0.jar ...
[info] 	[SUCCESSFUL ] org.flywaydb#flyway-core;4.0!flyway-core.jar (198ms)
```

**Comment by ksit** [comment]
@sourcesmith my bad for not providing all the details upfront.  I already have addSbtPlugin commented out in plugins.sbt, so it should not be the problem.

Having said that, the problem has been resolved.

**TL;DR** Just do what @axelfontaine suggested, and add  `https://flywaydb.org/repo` to your resolver list.

When I first modified the build, I tried to update my local Artifactory server to proxy the repo `https://flywaydb.org/repo`.  However, Artifactory gave me an error when I tried to test the connection.  I also tried hitting `https://flywaydb.org/repo` directly using a browser, and I got a 404 error.  So I didn't bother adding the repo to the resolver list -- and apparently this is **the** problem.

The repo is indeed working -- and you can test it using the following URL:

`https://flywaydb.org/repo/org/flywaydb/flyway-sbt_2.10_0.13/4.0/flyway-sbt-4.0.jar`

At the end, I just ignored the error from Artifactory, added the `https://flywaydb.org/repo` URL to the list of repository to proxy, and things are now working fine.

**Comment by fiadliel** [comment]
FYI for anybody who sees this in future, my issue was using http instead of https in the URL, which causes a "not found" error.

**Comment by getmetorajesh** [comment]
I am using `scala 2.11.8`, `SBT 0.13.13` and unable to download `flyway`. Its resolving to a non-existent url,

`https://flywaydb.org/repo/org/flywaydb/flyway-sbt_2.11_0.13/4.0.3/flyway-sbt-4.0.3.pom`
